front/app/containers/Admin/pagesAndMenu/containers/GenericHeroBannerForm/HeaderImageDropzone.test.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import HeaderImageDropzone from './HeaderImageDropzone';
+import { render, screen, waitFor, fireEvent } from 'utils/testUtils/rtl';
+
+jest.mock('utils/cl-intl');
+const file = new File(['file'], 'file.jpeg', {
+  type: 'image/jpeg',
+});
+
+describe('HeaderImageDropzone', () => {
+  it('renders properly', () => {
+    const component = render(
+      <HeaderImageDropzone
+        overlayColor={'#fff'}
+        overlayOpacity={90}
+        header_bg={null}
+        previewDevice="desktop"
+        layout="full_width_banner_layout"
+        onAdd={jest.fn()}
+        onRemove={jest.fn()}
+        headerError={null}
+      />
+    );
+    const dropZone = component.container.querySelector(
+      '#landingpage-header-dropzone'
+    );
+    expect(dropZone).toBeInTheDocument();
+  });
+
+  it('fires the onAdd handlers properly when an image is added and removed', async () => {
+    const onAdd = jest.fn();
+
+    const component = render(
+      <HeaderImageDropzone
+        overlayColor={'#fff'}
+        overlayOpacity={90}
+        header_bg={null}
+        previewDevice="desktop"
+        layout="full_width_banner_layout"
+        onAdd={onAdd}
+        onRemove={jest.fn()}
+        headerError={null}
+      />
+    );
+
+    const dropZone = component.container.querySelector(
+      '#landingpage-header-dropzone'
+    );
+
+    await waitFor(() => {
+      fireEvent.change(dropZone, {
+        target: { files: [file] },
+      });
+    });
+    await waitFor(() =>
+      expect(onAdd).toHaveBeenCalledWith([
+        expect.objectContaining({
+          base64: expect.any(String),
+          path: 'file.jpeg',
+        }),
+      ])
+    );
+  });
+
+  it('fires the onRemove handlers properly when an image is added and removed', async () => {
+    const onRemove = jest.fn();
+
+    const mockUploadFileParams = {
+      id: 'id',
+      filename: 'dog.jpeg',
+      base64: 'test',
+      url: 'test',
+      remote: true,
+      path: 'file.jpeg',
+    };
+
+    const mockUploadFile = {
+      ...file,
+      ...mockUploadFileParams,
+    };
+
+    render(
+      <HeaderImageDropzone
+        overlayColor={'#fff'}
+        overlayOpacity={90}
+        header_bg={[mockUploadFile]}
+        previewDevice="desktop"
+        layout="full_width_banner_layout"
+        onAdd={jest.fn()}
+        onRemove={onRemove}
+        headerError={null}
+      />
+    );
+
+    await waitFor(() => {
+      screen.getByRole('button').click();
+    });
+
+    await waitFor(() =>
+      expect(onRemove).toHaveBeenCalledWith(mockUploadFileParams)
```

Comment:
Hmm, I don't get this one. Calling the remove function sets header_bg to null (`bannerImageRemoveHandler` in `front/app/containers/Admin/pagesAndMenu/containers/HeroBanner/index.tsx`). Why do we need the call with mockUploadFileParams?

Reply:
This is just a test for the `HeaderImageDropzone` which imports `ImagesDropzone` which has this code:

```
  removeImage = (removedImage: UploadFile) => (event: React.FormEvent) => {
    event.preventDefault();
    event.stopPropagation();
    this.props.onRemove(removedImage);
  };
```

So it sends back the removed image, but I didn't look higher in the logic, we don't use that data, so it could be changed to just check that the function is called at all. I'll change it to just see that the function was called